src/main/java/org/threadly/util/StringUtils.java

```diff
@@ -37,26 +37,137 @@ private StringUtils() {
+   * @param minLength Minimum number of characters the returned string should be
+   * @param padChar Character to pad on to the start of to reach the minimum length
+   * @return A non-null string that is at least the length requested
+   */
+  public static String padStart(String sourceStr, int minLength, char padChar) {
+    sourceStr = nullToEmpty(sourceStr);
+    if (sourceStr.length() >= minLength) {
+      return sourceStr;
+    } else {
+      StringBuilder sb = new StringBuilder(minLength);
+      int padCount = minLength - sourceStr.length();
+      while (sb.length() < padCount) {
+        sb.append(padChar);
+      }
+      sb.append(sourceStr);
+      return sb.toString();
+    }
+  }
+
+  
+  /**
+   * Pads the end of the provided string with the provided character until a minimum length is 
+   * reached.  If the provided string is greater than or equal to the minLength it will be returned 
+   * without modification.  If the provided string is {@code null} it will be returned as an empty 
+   * string, padded to the minimum length.
+   * 
+   * @since 4.2.0
+   * @param sourceStr String to start from, this will be the start of the returned result string
+   * @param minLength Minimum number of characters the returned string should be
+   * @param padChar Character to pad on to the end of to reach the minimum length
+   * @return A non-null string that is at least the length requested
+   */
+  public static String padEnd(String sourceStr, int minLength, char padChar) {
+    sourceStr = nullToEmpty(sourceStr);
+    if (sourceStr.length() >= minLength) {
+      return sourceStr;
+    } else {
+      StringBuilder sb = new StringBuilder(minLength);
+      sb.append(sourceStr);
+      while (sb.length() < minLength) {
+        sb.append(padChar);
+      }
+      return sb.toString();
+    }
+  }
   
   /**
    * Produces a random string of the provided length.  This can be useful for unit testing, or any 
    * other time the string content is not important.  The returned string will be comprised of 
    * only alphanumeric characters.
    * 
+   * @deprecated Use {@link #makeRandomString(int)}
+   * 
    * @param length Number of characters the resulting string should be.
    * @return A string comprised of random characters of the specified length
    */
+  @Deprecated
   public static String randomString(int length) {
+    return makeRandomString(length);
+  }
+  
+  /**
+   * Produces a random string of the provided length.  This can be useful for unit testing, or any 
+   * other time the string content is not important.  The returned string will be comprised of 
+   * only alphanumeric characters.
+   * 
+   * @param length Number of characters the resulting string should be.
+   * @return A string comprised of random characters of the specified length
+   */
+  public static String makeRandomString(int length) {
      StringBuilder sb = new StringBuilder(length);
```

Comment:
Possibly, but this is primarily for unit testing anyways.  Not sure when random strings are useful otherwise (except I guess if you were making tokens).

I don't want to include that in this PR at least, but feel free to include a new one if you can benchmark it and show it to be better.